src/containers/MuppetAll.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { useState, useEffect } from 'react';
+import CreateMuppet from '../components/create/CreateMuppet';
+import MuppetList from '../components/muppets/MuppetList';
+import { getMuppets, postMuppet } from '../services/muppetApi';
+
+const MuppetAll = () => {
+  const [muppets, setMuppets] = useState([]);
+  //const history = useHistory();
+
+  useEffect(() => {
+    getMuppets()
+      .then(muppets => setMuppets(muppets));
+  }, []);
+
+  const handleChange = ({ target }) => {
+    setMuppets(target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    postMuppet(muppets)
+      //.then (() => history.push('/'))
+      .then(console.log);
+  };
+
+
+  return (
+    <div>
+      <CreateMuppet 
+        name="Kermit"
+        onChange={handleChange}
+        onSubmit={handleSubmit}/>
+      <MuppetList
+        { ...muppets } />
```

Comment:
muppetlist doesn't take any props and you do the same fetch in that component - it is better to do it here and pass it as a prop. You just need to change the component in the muppetlist file to take muppets as a prop